core/blockchain.go

```diff
@@ -1929,6 +1929,13 @@ func (bc *BlockChain) processBlock(block *types.Block, statedb *state.StateDB, s
 		return nil, err
 	}
 	vtime := time.Since(vstart)
+
+	// Commit the tries to in-memory buffers to finish populating any witnesses
+	if err = statedb.CommitWithoutFlush(bc.chainConfig.IsEIP158(block.Number())); err != nil {
+		return nil, err
+	}
+	// TODO(karalabe): Do the stateless cross validation here when merging it
+
```

Comment:
Hm what? We need the root to cross-validate, but afaict we don't obtain the root until we `Flush`. Which we just split off, as the second operation. So how does this help us?

Reply:
The root is already constant courtesy of IntermediareToor I guess. But in that case, the witness should also be complete?

Reply:
Root is already computed in `IntermediateRoot`. Commit just returns it again.